lib/DXIL/DxilShaderFlags.cpp

```diff
@@ -420,16 +438,22 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
   bool hasSamplerDescriptorHeapIndexing = false;
   bool hasAtomicInt64OnHeapResource = false;
 
-  // Used to determine whether to set ResMayNotAlias flag.
-  bool hasUAVs = M->GetUAVs().size() > 0;
+  bool hasUAVsGlobally = M->GetUAVs().size() > 0;
 
   bool hasAdvancedTextureOps = false;
-  bool hasWriteableMSAATextures = false;
+  bool hasWriteableMSAATexturesGlobal = false;
   bool hasSampleCmpGradientOrBias = false;
 
   bool hasWaveMMA = false;
   bool hasExtendedCommandInfo = false;
 
+  // UsesDerivatives is used to indicate any derivative use per-function, before
```

Comment:
Typo I think
```suggestion
  // hasDerivatives is used to indicate any derivative use per-function, before
```

Reply:
`UsesDerivatives` is the name of the flag, and intentional.  `hasDerivatives` is the name of the bool variable that we use locally to keep track of whether we found derivative usage, but may or may not result in the UsesDerivatives flag being set.